src/services/Api.php

```diff
@@ -58,4 +66,84 @@ public function getClient()
     {
         return $this->_client;
     }
+
+    /**
+     * @return array
+     */
+    public function getFromParams(): array
+    {
+        $storeLocation = Plugin::getInstance()->getAddresses()->getStoreLocationAddress();
+        return $this->_getAddressParams(self::TYPE_FROM, $storeLocation);
+    }
+
+    /**
+     * @param Address $address
+     * @return array
+     */
+    public function getToParams(Address $address): array
+    {
+        return $this->_getAddressParams(self::TYPE_TO, $address);
+    }
+
+    /**
+     * @param Order $order
+     * @param bool $includeAll
+     * @return array
+     */
+    public function getAmountsParams(Order $order, bool $includeAll = true): array
+    {
+        return [
+            'amount' => $order->getItemSubtotal() + $order->getTotalDiscount() + $order->getTotalShippingCost(),
+            'shipping' => $order->getTotalShippingCost(),
```

Comment:
I want to note that doing this here doesn't account for other adjusters. We have a shipping adjuster in our custom module — but this seems to run before that, and so this thinks shipping is $0 — causing us to not pay the correct amount of tax.

Reply:
Actually, I may have figured this out. So sorry for spamming this PR. Replacing the default shipping adjuster rather than just adding a new shipping adjuster did the trick.

Reply:
Hey @codyjames, sorry for the slow response! Yes, the tax refund handles partial refunds. I'm happy to jump on a call if you have lingering questions regarding this fork. If so, let me know how to best get in touch and we can set up a time.

Reply:
@jmauzyk Yes! A call would actually be very helpful. Want to send me an email at cody@humanshapes.co?